woocommerce/woocommerce-ios
# Store creation M2: show store summary after store creation success

Part of #8106 
⚠️ Please only review this PR when the base PR https://github.com/woocommerce/woocommerce-ios/pull/8107 is approved without critical suggestions


### Description


After domain selection and before IAP, we need to create a free site and then pass the site ID for purchasing the eCommerce plan via IAP. This PR integrates the site creation backend and UI, where the tap to continue a selected domain triggers the site creation API request and shows a spinner in the CTA at the same time. After site creation success, `StoreCreationCoordinator` navigates to a new screen added in this PR, `StoreCreationSummaryView`, wrapped in a hosting controller. In case of site creation failure, an error alert is shown (we can polish the error state later).

### Testing instructions


- Log in if needed
- Go to the Menu tab, and tap `Switch store`
- On the store picker, tap `+ Add a store`
- Tap `Create a new store` --> the domain selector should be shown
- Enter some valid query for domain suggestions
- Select a domain and tap Continue --> after a bit, the store summary screen should be shown as in the screenshots

### Screenshots


I tested the domain error path by hard-coding the domain to be an empty string in the site creation request.

domain selector | creating store | store summary | domain error
-- | -- | -- | --
![Simulator Screen Shot - iPhone 14 - 2022-11-14 at 15 03 52](https://user-images.githubusercontent.com/1945542/201595957-5e174c96-d1c1-4a35-881f-821078d33852.png) | ![Simulator Screen Shot - iPhone 14 - 2022-11-14 at 15 03 54](https://user-images.githubusercontent.com/1945542/201595974-0d08c1c9-f023-4615-ad4c-e08757a0b2ec.png) | ![Simulator Screen Shot - iPhone 14 - 2022-11-14 at 15 03 59](https://user-images.githubusercontent.com/1945542/201595977-4e58284d-fe29-48aa-954c-38a87d2e63e3.png) | ![Simulator Screen Shot - iPhone 14 - 2022-11-14 at 14 35 31](https://user-images.githubusercontent.com/1945542/201596001-172d161e-326c-4a56-bb7f-0151dbfa3ace.png)


---
